dpctl/tensor/libtensor/include/kernels/elementwise_functions/expm1.hpp

```diff
@@ -73,6 +74,45 @@ template <typename argT, typename resT> struct Expm1Functor
             const realT x = std::real(in);
             const realT y = std::imag(in);
 
+            // special cases
+            if (std::isinf(x)) {
+                if (x > realT(0)) {
+                    // positive infinity cases
+                    if (!std::isfinite(y)) {
+                        return resT{x, std::numeric_limits<realT>::quiet_NaN()};
+                    }
+                    else if (y == realT(0)) {
+                        return in;
+                    }
+                    else {
+                        return (x * resT{std::cos(y), std::sin(y)});
```

Comment:
I think we should do instead, to ensure that `NaN` does not come up in product computations:

```suggestion
                        return resT{std::copysign(x, std::cos(y)), std::copysign(x, std::sin(y))});
```

Reply:
Yes, this makes sense. I'll make this change.

Reply:
Other than this change, the PR looks good. I have built and tested it on Iris Xe and PVC.